frodo.js

```diff
@@ -1,55 +1,55 @@
 /**
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.